govet cmd/tooken_user.go: the cancel function returned by context.WithTimeout should be called

**Do you want to request a *feature* or report a *bug*?**
bug

**What is the current behavior?**
```
cd cmd
go vet
...
token_user.go:121:9: the cancel function returned by context.WithTimeout should be called, not discarded, to avoid a context leak (govet)
			ctx, _ := context.WithTimeout(context.Background(), time.Second*5)
			     ^
``` 

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**

**What is the expected behavior?**
go vet passes

**Which version of the software is affected?**
beta-9

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -19,7 +19,7 @@ jobs:
       - checkout
       - run: curl -L https://git.io/vp6lP | sh
       - run: mv ./bin/* $GOPATH/bin
-      - run: gometalinter --disable-all --enable=gofmt --vendor ./...
+      - run: gometalinter --disable-all --enable=gofmt --enable=vet --vendor ./...
 
   test:
     docker:

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -17,6 +17,11 @@ jobs:
 #            echo 'export GO111MODULE=on' >> $BASH_ENV
 #            source $BASH_ENV
       - checkout
+      - run:
+          name: Enable go1.11 modules
+          command: |
+            echo 'export GO111MODULE=on' >> $BASH_ENV
+            source $BASH_ENV
       - run: curl -L https://git.io/vp6lP | sh
       - run: mv ./bin/* $GOPATH/bin
       - run: go mod download

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -19,6 +19,7 @@ jobs:
       - checkout
       - run: curl -L https://git.io/vp6lP | sh
       - run: mv ./bin/* $GOPATH/bin
+      - run: go mod download
       - run: gometalinter --disable-all --enable=gofmt --enable=vet --vendor ./...
 
   test:

--- a/cmd/server/handler_oauth2_factory.go
+++ b/cmd/server/handler_oauth2_factory.go
@@ -118,7 +118,7 @@ func newOAuth2Provider(c *config.Config) fosite.OAuth2Provider {
 	}
 
 	if tracer, err := c.GetTracer(); err == nil && tracer.IsLoaded() {
-		hasher = &tracing.TracedBCrypt{fc.HashCost}
+		hasher = &tracing.TracedBCrypt{WorkFactor: fc.HashCost}
 	}
 
 	return compose.Compose(

--- a/cmd/token_user.go
+++ b/cmd/token_user.go
@@ -120,7 +120,8 @@ var tokenUserCmd = &cobra.Command{
 		server := &http.Server{Addr: fmt.Sprintf(":%d", port), Handler: r}
 		var shutdown = func() {
 			time.Sleep(time.Second * 1)
-			ctx, _ := context.WithTimeout(context.Background(), time.Second*5)
+			ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+			defer cancel()
 			server.Shutdown(ctx)
 		}
 

--- a/config/config.go
+++ b/config/config.go
@@ -330,7 +330,7 @@ func (c *Config) Context() *Context {
 	}
 
 	if tracer, err := c.GetTracer(); err == nil && tracer.IsLoaded() {
-		hasher = &tracing.TracedBCrypt{c.BCryptWorkFactor}
+		hasher = &tracing.TracedBCrypt{WorkFactor: c.BCryptWorkFactor}
 	}
 
 	c.context = &Context{

